homeassistant/components/mobile_app/websocket_api.py

```diff
@@ -0,0 +1,85 @@
+"""Websocket API for mobile_app."""
+from aiohttp.http_websocket import WSMessage
+
+from homeassistant.components.cloud import (async_delete_cloudhook,
+                                            async_is_logged_in)
+from homeassistant.components.websocket_api import (ActiveConnection,
+                                                    async_register_command,
+                                                    async_response,
+                                                    error_message,
+                                                    result_message,
+                                                    websocket_command,
+                                                    ws_require_user)
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.typing import HomeAssistantType
+
+from .const import (ATTR_DELETED_IDS, ATTR_REGISTRATIONS, ATTR_STORE,
+                    CONF_USER_ID, CONF_WEBHOOK_ID, DOMAIN,
+                    SCHEMA_WS_DELETE_REGISTRATION, SCHEMA_WS_GET_REGISTRATION)
+
+from .helpers import safe_device, savable_state
+
+
+def register_websocket_handlers(hass: HomeAssistantType) -> bool:
+    """Register the websocket handlers."""
+    async_register_command(hass, websocket_get_registration)
+
+    async_register_command(hass, websocket_delete_registration)
+
+    return True
+
+
+@ws_require_user()
+@async_response
+@websocket_command(SCHEMA_WS_GET_REGISTRATION)
+async def websocket_get_registration(
+        hass: HomeAssistantType, connection: ActiveConnection,
+        msg: WSMessage) -> None:
```

Comment:
This is not a WSMessage, it's a dict.

Reply:
Ah, it must become a `dict` when it's run through validation then.